src/main/java/bio/terra/service/snapshot/flight/create/AuthorizeSnapshot.java

```diff
@@ -74,6 +79,14 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
             List<String> fileIds = fireStoreDao.getDatasetSnapshotFileIds(dataset, snapshotId.toString());
             gcsPdao.setAclOnFiles(dataset, fileIds, readersPolicyEmail);
         }
+        // add a fault that forces an exception to make sure the undo works
+        try {
+            configService.fault(ConfigEnum.CREATE_SNAPSHOT_FAULT, () -> {
+                throw new RuntimeException("fault insertion");
+            });
+        } catch (Exception e) {
```

Comment:
What is the try-catch for?
I don't think you need it.

Reply:
well I had to move files, but Dan, I think I do need the try-catch, because the configService.fault() requires an Exception and the doStep implements an interface that doesnt.....